src/shared/regions/defaultRegionProvider.ts

```diff
@@ -29,14 +27,20 @@ export class DefaultRegionProvider implements RegionProvider {
         return this.onRegionProviderUpdatedEmitter.event
     }
 
-    public async getRegionData(): Promise<RegionInfo[]> {
-        return this._loadedRegions
-    }
-
     public isServiceInRegion(serviceId: string, regionId: string): boolean {
         return !!this.regionIdToRegionData.get(regionId)?.serviceIds.find(x => x === serviceId) ?? false
     }
 
+    public getPartitionId(regionId: string): string | undefined {
+        return this.regionIdToRegionData.get(regionId)?.partitionId ?? undefined
```

Comment:
In what cases would we expect this to return undefined? Is this serious enough that we should log it or throw an error?

Reply:
If the user enters an invalid region, or one that is not reflected in the endpoints file. This behaviour allows callers to decide what to do, which presently they default to the most common partition ('aws'). This seems like a reasonable approach until/unless anecdotes tell us otherwise.

Reply:
Would a log be out of place here? We should capture this, whether it be here or on the caller end. Just making sure, this would 1: get fired if a user has a bad region set in `.aws/config` and 2: won't affect any other calls in the future?

If this affects other calls down the line, we'll definitely want to handle it as an error.

Reply:
I will add a log to help inform on frequency, and assist with unexpected scenarios that may get reported by users.